pytorch_lightning/trainer/supporters.py

```diff
@@ -120,26 +120,124 @@ def mean(self):
 
 
 class PredictionCollection(object):
-    def __init__(self, global_rank: int, world_size: int):
+
+    ID_KEY = 'id'
+
+    def __init__(self, global_rank: int, world_size: int, all_gather: Callable):
```

Comment:
Might be a good idea changing this to `all_gather_fn` to make it clear this is an op

Reply:
Good call !